### ISSUE TITLE
github - Login results in Error message

### ISSUE THREAD

**Issue by makitso** [fisharebest/webtrees#2876]
Select individual from home page.
Then select Sign in.
Error page is displayed.  However, if you hit the browser backspace you are indeed logged in.
This may be similar to issue #2872

![image](https://user-images.githubusercontent.com/5147603/70805583-7a189800-1d76-11ea-934e-5af9814f6e46.png)

**Comment by fisharebest** [comment]
I just tried logging in from this URL - with Firefox and Chrome - and didn't see a problem.

https://dev.webtrees.net/demo-dev/tree/demo/individual/i1/Queen%20Elizabeth%20II

Can you enable dev tools in your browser, and tell me what URL?
i.e. is it redirecting back to the same URL, or is it flipping between two?

**Comment by makitso** [comment]
URL from individual
 http://localhost/?route=%2Ftree%2Fpeters.ged%2Findividual%2FI138%2FArlene%2520Dorothy%2520Otte/index.php?route=0
URL from Sign on
https://www.webtrees.net/index.php/en/forum/help-for-2-0/33793-too-many-redirects#73430

![image](https://user-images.githubusercontent.com/5147603/70814932-c5887180-1d89-11ea-899d-4078c35f262e.png)

**Comment by makitso** [comment]
Perhaps the detail headers:
[HTTP Headers.txt](https://github.com/fisharebest/webtrees/files/3961725/HTTP.Headers.txt)

**Comment by fisharebest** [comment]
```
GET /?route=%2Ftree%2Fpeters.ged%2Findividual%2FI138%2FArlene%2520Dorothy%2520Otte 

Location: http://localhost/?route=%2Ftree%2Fpeters.ged%2Findividual%2FI138%2FArlene%2520Dorothy%2520Otte/index.php?route=0
```

The request to `/?route=...` is wrong.  It should be `/index.php?route=...`

What value do you have for `base_url` in `data/config.ini.php`?

**Comment by makitso** [comment]
I knew you were going to ask me that.  I do not have the base_url set

**Comment by makitso** [comment]
I knew you were going to ask me that.  I do not have the base_url set

**Comment by makitso** [comment]
I knew you were going to ask me that.  I do not have the base_url set

**Comment by makitso** [comment]
So, the problem does not exist if I add base_url="http://localhost" to my config.ini.php file.  
But, my question is  would the base_url value find it's way into the config file, per an upgrade, etc?  My point is, with an older installation of webtrees, this value is not by default or update, in the config.ini.php file.

**Comment by fisharebest** [comment]
> So, the problem does not exist if I add base_url="http://localhost" to my config.ini.php file.

The installer for 2.0 sets this value.

> But, my question is would the base_url value find it's way into the config file, per an upgrade, etc? 

Yes.  The next release of 1.7 will do this.
The code has been written, but is not yet fully tested.

It will therefore be necessary to upgrade to >=1.7.16 before upgrading to 2.0

**Comment by makitso** [comment]
So, people who upgrade from 1.7 to 2.0 will have their config.ini.php files updated?
What about people on early versions of 2.0, that have been progressively updating from Alpha to Beta releases and then to 2.0.  How will this change be made for them?

**Comment by fisharebest** [comment]
If you do not include the `base_url` setting, then the library does some auto-detecting.

It works in many cases.  It worked on my laptop, and on the dev server.

But it is dependent on information passed by the webserver to PHP, which may be incomplete or incorrect.

I suspect something is not quite right in your webserver config.

> Alpha to Beta releases and then to 2.0

According to the statistics, nobody is using anything earlier than beta.3

So anyone in this category has already upgraded, and nobody has reported an issue.

**Comment by xmlf** [comment]
If base_url  not set, an infinite loop redirection will occur.
I am preparing to report this issue.

**Comment by fisharebest** [comment]
> If base_url not

`base_url` is set during the installation.

If you delete parts of the code, it will stop working.

@makitso - do you have URL rewriting enabled?  I can't think of another reason you might have URLs without `index.php`.

**Comment by makitso** [comment]
@fisharebest  - adding a base_url to my locahost and production site resolved the issue.

> do you have URL rewriting enabled?
I have this in my .htaccess file
RewriteEngine On
RewriteCond %{HTTPS} !=on
RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

**Comment by gregkuhnert** [comment]
base_url was not set for me during installation. I did not remove it. But indeed that is a fix for the issue. I did not have pretty URL's enabled either. It was just a clean install, and it was broken.

Speaking of which, the README file should be updated with instructions for pretty URL activation - I had to go digging, where this should really be in the documentation in the release archive IMHO.
